backend/canisters/community/impl/src/timer_job_types.rs

```diff
@@ -130,3 +149,43 @@ impl Job for MarkGroupImportCompleteJob {
         mark_import_complete(self.group_id, self.channel_id);
     }
 }
+
+impl Job for ClosePrizeJob {
+    fn execute(&self) {
+        if let Some(pending_transaction) = mutate_state(|state| {
+            if let Some(channel) = state.data.channels.get_mut(&self.channel_id) {
+                channel
+                    .chat
+                    .events
+                    .close_prize(self.thread_root_message_index, self.message_index, state.env.now_nanos())
+            } else {
+                None
+            }
+        }) {
+            let make_transfer_job = MakeTransferJob { pending_transaction };
+            make_transfer_job.execute();
+        }
+    }
+}
+
+impl Job for MakeTransferJob {
+    fn execute(&self) {
+        let sender = read_state(|state| state.env.canister_id());
+        let pending = self.pending_transaction.clone();
+        ic_cdk::spawn(make_transfer(pending, sender));
+
+        async fn make_transfer(pending_transaction: PendingCryptoTransaction, sender: CanisterId) {
+            if let Err(error) = process_transaction(pending_transaction.clone(), sender).await {
+                error!(?error, "Transaction failed");
+                mutate_state(|state| {
+                    let now = state.env.now();
+                    state.data.timer_jobs.enqueue_job(
+                        TimerJob::MakeTransfer(MakeTransferJob { pending_transaction }),
+                        now + HOUR_IN_MS,
```

Comment:
I reckon an hour is too long. The main cause of failure would be the subnet being upgraded which only lasts a few minutes. So maybe retry after 1 min?

Reply:
changed